### PR TITLE
Closes #2582 update derive_vars_period

### LINKED ISSUE #2582: Bug: derive_vars_period fails when there is only one new var
### What happened?

derive_vars_period fails when there are only one new_vars

### Session Information
```
R version 4.3.3 (2024-02-29)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Ubuntu 22.04.4 LTS

Matrix products: default
BLAS:   /usr/lib/x86_64-linux-gnu/openblas-pthread/libblas.so.3 
LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.20.so;  LAPACK version 3.10.0

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

time zone: Etc/UTC
tzcode source: system (glibc)

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] admiral_1.1.1.9030 tidyr_1.3.0        testthat_3.2.0    

loaded via a namespace (and not attached):
 [1] htmlwidgets_1.6.4 devtools_2.4.5    remotes_2.5.0     processx_3.8.2   
 [5] callr_3.7.3       vctrs_0.6.4       tools_4.3.3       ps_1.7.5         
 [9] generics_0.1.3    tibble_3.2.1      fansi_1.0.5       pkgconfig_2.0.3  
[13] desc_1.4.2        lifecycle_1.0.3   compiler_4.3.3    stringr_1.5.0    
[17] brio_1.1.3        httpuv_1.6.15     htmltools_0.5.8.1 usethis_2.2.2    
[21] later_1.3.2       pillar_1.9.0      crayon_1.5.2      urlchecker_1.0.1 
[25] ellipsis_0.3.2    cachem_1.1.0      sessioninfo_1.2.2 mime_0.12        
[29] tidyselect_1.2.0  digest_0.6.33     stringi_1.7.12    dplyr_1.1.3      
[33] purrr_1.0.2       rprojroot_2.0.3   fastmap_1.2.0     cli_3.6.3        
[37] magrittr_2.0.3    pkgbuild_1.4.2    utf8_1.2.3        withr_2.5.1      
[41] admiraldev_1.1.0  diffdf_1.0.4      prettyunits_1.2.0 waldo_0.5.3      
[45] promises_1.3.0    lubridate_1.9.3   timechange_0.2.0  hms_1.1.3        
[49] memoise_2.0.1     shiny_1.9.1       evaluate_0.22     miniUI_0.1.1.1   
[53] profvis_0.3.8     rlang_1.1.1       Rcpp_1.0.11       xtable_1.8-4     
[57] glue_1.8.0        pkgload_1.3.3     rstudioapi_0.15.0 R6_2.5.1         
[61] fs_1.6.3     
```
### Reproducible Example
```
expected <- tibble::tribble(
    ~USUBJID, ~AP01SDT,
    "1",      "2021-01-04",
    "2",      "2021-02-02",
  ) %>%
    mutate(
      across(matches("AP\\d\\d[ES]DT"), ymd)
    ) %>%
    mutate(
      STUDYID = "xyz"
    )

  period_ref <- tibble::tribble(
    ~USUBJID, ~APERIOD, ~APERSDT,     ~APEREDT,     ~APERSDTM,             ~APEREDTM,
    "1",             1, "2021-01-04", "2021-02-06", "2021-01-04 10:01:02", "2021-02-06 14:03:04",
    "2",             1, "2021-02-02", "2021-03-02", "2021-02-02 04:11:34", "2021-03-02 16:55:59"
  ) %>%
    mutate(
      STUDYID = "xyz",
      APERIOD = as.integer(APERIOD),
      APERSDT = ymd(APERSDT),
      APEREDT = ymd(APEREDT),
      APERSDTM = ymd_hms(APERSDTM),
      APEREDTM = ymd_hms(APEREDTM)
    )

  adsl <- tibble::tibble(STUDYID = "xyz", USUBJID = c("1", "2"))

derive_vars_period(
      adsl,
      dataset_ref = period_ref,
      new_vars = exprs(APxxSDT = APERSDT)
    )
```

### PR DESCRIPTION
**Pull request by clarkliming**
Closes #2582 

Thank you for your Pull Request! We have developed this task checklist from the [Development Process Guide](https://pharmaverse.github.io/admiral/CONTRIBUTING.html#detailed-development-process) to help with the final steps of the process. Completing the below tasks helps to ensure our reviewers can maximize their time on your code as well as making sure the admiral codebase remains robust and consistent.   

Please check off each taskbox as an acknowledgment that you completed the task or check off that it is not relevant to your Pull Request. This checklist is part of the Github Action workflows and the Pull Request will not be merged into the `main` branch until you have checked off each task.

- [x] Place Closes #<insert_issue_number> into the beginning of your Pull Request Title (Use Edit button in top-right if you need to update)
- [x] Code is formatted according to the [tidyverse style guide](https://style.tidyverse.org/). Run `styler::style_file()` to style R and Rmd files
- [x] Updated relevant unit tests or have written new unit tests, which should consider realistic data scenarios and edge cases, e.g. empty datasets, errors, boundary cases etc. - See [Unit Test Guide](https://pharmaverse.github.io/admiraldev/articles/unit_test_guidance.html#tests-should-be-robust-to-cover-realistic-data-scenarios)
- [x] If you removed/replaced any function and/or function parameters, did you fully follow the [deprecation guidance](https://pharmaverse.github.io/admiraldev/articles/programming_strategy.html#deprecation)?
- [x] Review the [Cheat Sheet](https://github.com/pharmaverse/admiral/blob/main/inst/cheatsheet/admiral_cheatsheet.pdf). Make any required updates to it by editing the file `inst/cheatsheet/admiral_cheatsheet.pptx` and re-upload a PDF and a PNG version of it to the same folder. (The PNG version can be created by taking a screenshot of the PDF version.)
- [x] Update to all relevant roxygen headers and examples, including keywords and families. Refer to the [categorization of functions](https://pharmaverse.github.io/admiraldev/articles/programming_strategy.html#categorization-of-functions) to tag appropriate keyword/family.
- [x] Run `devtools::document()` so all `.Rd` files in the `man` folder and the `NAMESPACE` file in the project root are updated appropriately
- [x] Address any updates needed for vignettes and/or templates
- [x] Update `NEWS.md` under the header `# admiral (development version)` if the changes pertain to a user-facing function (i.e. it has an `@export` tag) or documentation aimed at users (rather than developers). A Developer Notes section is available in `NEWS.md` for tracking developer-facing issues.  
- [x] Build admiral site `pkgdown::build_site()` and check that all affected examples are displayed correctly and that all new functions occur on the "[Reference](https://pharmaverse.github.io/admiral/reference/index.html)" page. 
- [x] Address or fix all lintr warnings and errors - `lintr::lint_package()`
- [x] Run `R CMD check` locally and address all errors and warnings - `devtools::check()`
- [x] Link the issue in the Development Section on the right hand side.
- [x] Address all merge conflicts and resolve appropriately
- [x] Pat yourself on the back for a job well done! Much love to your accomplishment!

### CODE REVIEW

**Review by bundfussr — CHANGES_REQUESTED** [review]
Thanks for reporting the bug and providing a fix!

File: tests/testthat/test-period_dataset.R [hunk 1]
```diff
@@ -433,3 +433,44 @@ test_that("derive_vars_period Test 13: periods", {
     keys = c("USUBJID")
   )
 })
+
+## Test 14: Periods derived even when only when one period variable needed ----
+test_that("derive_vars_period Test 13: periods", {
```
[the remark this answers is not in the document]
Comment (clarkliming) [reply tests/testthat/test-period_dataset.R]: I updated my part; for other part I keep the original lines (as the function did not correctly spot the function being tested)

**Review by bundfussr — COMMENTED** [review]
Thanks @clarkliming , looks good now.

Please resolve the merge conflict. Then it is ready to merge.

**Comment by clarkliming** [comment]
> Thanks @clarkliming , looks good now.
> 
> Please resolve the merge conflict. Then it is ready to merge.

conflict resolved now

**Review by bundfussr — APPROVED** [review]

**Pull request merged**